### ISSUE TITLE
OAuth2 client  : missing calls to ungetOAuthService and deleteServiceAndAccessToken

### ISSUE THREAD

**Issue by lolodomo** [openhab/openhab-addons#14818]
17 bindings use the OAuth2 client provided by the OH core framework.

Only 9 are calling `ungetOAuthService`. It could be missing for the following bindings:

- hydrawise
- livisismarthome
- mercedesmen
- mielecloud
- myq
- nest
- googlestt
- googletts.

Only ecobee and mielecloud bindings are calling `deleteServiceAndAccessToken`. It could be missing for the following bindings:

- automower
- boschindego
- bticinosmarther
- ecowatt
- groupepsa
- homeconnect
- hydrawise
- livisismarthome
- mercedesmen
- myq
- nest
- spotify
- webexteams
- googlestt
- googletts.

**Comment by lolodomo** [comment]
@jlaur : I should update again all bindings (when necessary) because `deleteServiceAndAccessToken` has to be called in `handleRemoval` even if no OAuth service was initialized. It is necessary to remove the entries from the storage.

**Comment by mhilbush** [comment]
@lolodomo The ecobee binding calls `deleteServiceAndAccessToken` when the Ecobee cloud forces a reauthorization, but not when the thing is removed. I'll submit a PR that calls `deleteServiceAndAccessToken` when the thing is removed.

**Comment by lolodomo** [comment]
@jlaur : it is a pity, we did not update the two last ones: bticinosmarther and mielecloud.

**Comment by jlaur** [comment]
> @jlaur : it is a pity, we did not update the two last ones: bticinosmarther and mielecloud.

Yes. For mielecloud I created #14783, but this could be done separately.

**Comment by lsiepel** [comment]
@lolodomo only a few bindings are left, besides time, any specific reason they are not checked/fixed yet?

**Comment by lolodomo** [comment]
> @lolodomo only a few bindings are left, besides time, any specific reason they are not checked/fixed yet?

I suppose @jlaur and me switched to other PRs.

**Comment by jlaur** [comment]
@lolodomo - I've marked mielecloud as done on the list after #18264 was merged.
